AWSPinpoint/AWSPinpointAnalyticsClient.m

```diff
@@ -393,16 +393,20 @@ -(void) removeGlobalMetricForKey:(NSString*) theKey
     }
 }
 
-- (void) setCampaignAttributes:(NSDictionary*) campaign {
-    _globalCampaignAttributes = campaign;
-    [self.eventRecorder updateSessionStartWithCampaignAttributes:campaign];
+- (void) setFeatureAttributes:(NSDictionary*) attributes {
+    // Remove previous global feature attributes from _globalAttributes
+    // This is to prevent _globalAttributes containing attributes from multiple features (campaign/journey)
+    [self removeAllGlobalFeatureAttributes];
```

Comment:
This is necessary because these attributes are added to both _globalAttributes and _globalCampaign attributes in `AWSPinpointNotificationManager::addGlobalPinpointMetadataForNotification`
It seems redundant to me but I'm sure it was there for a reason. I can add unit test

Reply:
The main reason I don't like this is because by calling `setFeatureAttributes ` we are causing a side effect of removeAllGlobalFeatureAttributes.  This doesn't feel right.

So, the main question comes back to is: What is the main use case of calling `setFeatureAttributes:` ?  It seems like the AWSPinpointNotificationManager is calling it from `handleNotificationReceived:withNotification:shouldHandleNotificationDeepLink` and
`interceptDidFinishLaunchingWithOptions`, and by adding a call to `removeAllGlobalFeatureAttributes` it seems that the concern is that you can get one notification with journey information, and another notification with campaign information?   And if we get these two payloads in succession, it's possible for keys from the journey to "bleed" into the one with campaign information?

Can you help me better understand this so we can come up with a better solution?